File: 07-agent/cc-agent/src/server/routes/agent.ts
import { Hono } from "hono";
import { streamSSE } from "hono/streaming";
import { z } from "zod";
import type { AgentRunParams } from "../../agent/model.js";
import { ChannelId } from "../../channels/types.js";
import type { ServerContext } from "../context.js";

const AgentMessageSchema = z.object({
  channelId: z.nativeEnum(ChannelId).default(ChannelId.Wechat),
  appId: z.string().default("local"),
  groupId: z.string().default("local"),
  threadId: z.string().optional(),
  senderId: z.string().optional(),
  senderName: z.string().optional(),
  replyTo: z.string().optional(),
  content: z.string().min(1),
  isPrivate: z.boolean().default(true)
});

export function createAgentRoutes(ctx: ServerContext) {
  const app = new Hono();

  app.post("/messages", async (c) => {
    const message = toAgentRunParams(AgentMessageSchema.parse(await c.req.json()));
    const result = await ctx.agent.run(message);
    return c.json(result);
  });

  app.post("/messages/stream", async (c) => {
    const message = toAgentRunParams(AgentMessageSchema.parse(await c.req.json()));

    return streamSSE(c, async (stream) => {
      await ctx.agent.run(message, {
        stream: true,
        onEvent: async (event) => {
          await stream.writeSSE({
            event: event.type,
            data: JSON.stringify(event)
          });
        }
      });
    });
  });

  return app;
}

function toAgentRunParams(input: z.infer<typeof AgentMessageSchema>): AgentRunParams {
  return {
    ...input,
    source: "user"
  };
}

File: 07-agent/cc-agent/app/gateway/routes/agent.py
from typing import Any

from fastapi import APIRouter, Request
from pydantic import AliasChoices, BaseModel, ConfigDict, Field

from app.agent.model import (
    AgentRunParams,
    AgentRunSource,
    ChannelId,
    NullAgentRunContext,
)

router = APIRouter()


class AgentRunRequest(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    channel_id: ChannelId = Field(validation_alias=AliasChoices("channelId", "channel_id"))
    app_id: str = Field(validation_alias=AliasChoices("appId", "app_id"))
    group_id: str = Field(validation_alias=AliasChoices("groupId", "group_id"))
    content: str = Field(validation_alias=AliasChoices("content", "text"))
    is_private: bool = Field(default=False, validation_alias=AliasChoices("isPrivate", "is_private"))
    source: AgentRunSource = "user"
    thread_id: str | None = Field(default=None, validation_alias=AliasChoices("threadId", "thread_id"))
    sender_id: str | None = Field(default=None, validation_alias=AliasChoices("senderId", "sender_id", "user_id"))
    sender_name: str | None = Field(default=None, validation_alias=AliasChoices("senderName", "sender_name"))
    reply_to: str | None = Field(
        default=None,
        validation_alias=AliasChoices("replyTo", "reply_to", "channelMessageId", "channel_message_id"),
    )


@router.post("/run")
async def run_agent(request: Request, body: AgentRunRequest) -> dict[str, Any]:
    params = AgentRunParams(
        channel_id=body.channel_id,
        app_id=body.app_id,
        group_id=body.group_id,
        thread_id=body.thread_id,
        sender_id=body.sender_id,
        sender_name=body.sender_name,
        reply_to=body.reply_to,
        content=body.content,
        is_private=body.is_private,
        source=body.source,
    )
    return await request.app.state.agent_session.run(params, NullAgentRunContext())
